addon/utils/query-cache.js

```diff
@@ -0,0 +1,69 @@
+
+const CACHE_DECAY_TIMEOUT = 5000;
+
+export default class QueryCache {
+  constructor() {
+    this._internalCache = Object.create(null);
+    this.decayId = null;
+  }
+
+  add(type, key, value) {
+    if (!key || !value) {
+      return;
+    }
+
+    if (!this._internalCache[type]) {
+      this._internalCache[type] = Object.create(null);
```

Comment:
Is there a reason to have a nested object here? It may be to evict by type, but I don't see the code for that (but I may just have missed it) My recommendation would be to keep it flat, but combine keys:

```js
key = type + '!'  + key
```